db/migrate/20150227185631_add_users_state_flag.rb

```diff
@@ -0,0 +1,20 @@
+class AddUsersStateFlag < ActiveRecord::Migration
+  def up
+    add_column :users, :state, :string, default: 'temp'
+    execute "update users set state='temp' where is_temp='t'"
+    execute "update users set state='activated' where is_temp='f'"
+    # SQLite lacks support for the alter column statement
+    unless "SQLite" == ActiveRecord::Base.connection.adapter_name
+      execute "alter table users alter column state set not null"
+    end
+    remove_column :users, :is_temp
```

Comment:
I think if you specify a default, though, you're fine? It should just default all rows to the default value.